test/data.jl

```diff
@@ -262,13 +262,13 @@ module TestData
 
     # # TODO: Restore this functionality
     # m1 = join(dt1, dt2, on = [:a,:b])
-    # @test isequal(m1[:a], Vector(["x", "x", "y", "y", fill("x", 5)]))
+    # @test m1[:a] == Vector(["x", "x", "y", "y", fill("x", 5)]))
```

Comment:
@cjprybol, do you know what else needs to happen here? I mean, doing the actual `join` doesn't produce an error on this branch, but I'm not sure if there's something inherently wrong with what we're doing. As far as I can tell, we're doing a join on the `a` and `b` columns, which has duplicate rows in `dt1` and it looks like we're just joining the single `x A` row from `dt2` instead of joining all the `x A` rows in `dt2` to each `x A` in `dt1`. Is that the right direction?